docker/runtime/ruby-2.4/http-trigger/kubeless.rb

```diff
@@ -1,33 +1,49 @@
+# Copyright (c) 2016-2017 Bitnami
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
 #!/usr/bin/env ruby
+require 'sinatra'
 
-
-modName = ENV['MOD_NAME']
-funcHandler = ENV['FUNC_HANDLER']
-
-
-modRootPath = ENV['MOD_ROOT_PATH'] ? ENV['MOD_ROOT_PATH'] : '/kubeless/'
-modPath = [ modRootPath, modName, '.rb' ]
-puts('Loading', modPath.join())
+# Don't buffer stdout
+$stdout.sync = true
+MOD_NAME = ENV['MOD_NAME']
+FUNC_HANDLER = ENV['FUNC_HANDLER']
+MOD_ROOT_PATH = ENV.fetch('MOD_ROOT_PATH', '/kubeless/')
+MOD_PATH = "#{File.join(MOD_ROOT_PATH, MOD_NAME)}.rb"
 
 begin
-  require 'sinatra'
-  require  modPath.join()
-rescue RuntimeError => e
-  puts('No valid module found for the name: lambda, Failed to import module')
-  exit 1
+  puts "Loading #{MOD_PATH}"
+  mod = Module.new
```

Comment:
Why do you need to create a module on the fly? I believe you can just load the file using `require_relative` to load the file and you can `send` the method directly:

```
# test.rb
def hello
  puts 'hello'
end

# On an IRB session
$ irb
irb(main):001:0> require_relative './test.rb'
=> true
irb(main):002:0> send(:hello)
hello
=> nil
irb(main):003:0> hello
hello
=> nil
```

If you want to check if the method exists, you can call `method` to check if the `FUNC_HANDLER` is defined:

```
begin
  method(FUNC_HANDLER)
rescue NameError => e
  raise e, "No valid function found for the name: #{MOD_NAME}.#{FUNC_HANDLER}, failed to import module"
end
```

Reply:
Maybe it's a bit defensive but I wanted to isolate the function to its own module.

Reply:
In that case, it's fine :)